lib/lotus/model/adapters/sql_adapter.rb

```diff
@@ -230,15 +230,17 @@ def connection_string
         # Executes raw sql directly on the connection
         #
         # @param raw [String] the raw sql statement to execute on the connection
+        # @param block [Proc] code to be executed directly in the adapter context, with that you can
+        # manipulate low level ResultSet objects.
         #
         # @return [Object]
         #
         # @raise [Lotus::Model::InvalidQueryError] if raw statement is invalid
         #
         # @since 0.3.1
-        def execute(raw)
+        def execute(raw, &block)
```

Comment:
I didn't forget this comment, I'll give it special attention soon when I get time ;)

Very interesting, this PR deals with what I questioned here:

>When using execute directly we're adrift of the adapter, when it is
the Ruby SQlite3 adapter we get SQlite3::ResultSet as result,
but when dealing with Java JDBC SQLite adapter with get
Java::OrgSqliteJdbc4::JDBC4ResultSet which is a low level object that
doesn't have the Ruby interface that we expect.

> Facing that we need to question ourselves if we want to expose the raw
adapter reponse or a object with a cross-platform normalized interface.

The `fetch` method tries to normalize cross-adapter interface :)